project-backend/Providers/JobProvider/JobProvider.cs

```diff
@@ -33,16 +35,24 @@ public JobDAO AddJob(string name, string description, int userId)
 
         public JobDAO GetJobById(int jobId)
         {
-            JobDAO requiredJob;
             try
             {
-                requiredJob = _dbContext.Jobs.AsQueryable().Where(job => job.Id == jobId).First();
+                JobDAO requiredJob = _dbContext.Jobs.Include(job => job.Images).Where(job => job.Id == jobId).First();
+                return requiredJob;
             }
```

Comment:
Had to include the Images, otherwise it produced a nasty null exception.

Reply:
Nice catch, this didn't appear in tests because we were mocking the provider :))